### ISSUE TITLE
[HOLD for payment 2024-10-16] [HOLD for payment 2024-10-14] [HOLD for payment 2024-10-11] [$250] Task - Long Task title is overlapped by task arrow

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#49086]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.33-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4955214&group_by=cases:section_id&group_order=asc&group_id=291935
**Email or phone of affected tester (no customers):** gocemate+a2096@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Go to Group chat> +> Assign Task
2. In title field enter long text> Finish the flow
3. Take a look at Task title in Group chat

## Expected Result:
Task title should not be overlapped by arrow

## Actual Result:
Task title is overlapped by task arrow

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/a56f5ff7-f70f-466b-af99-89959042df96

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021834296909071158877</li>
        <li>Upwork Job ID: 1834296909071158877</li>
        <li>Last Price Increase: 2024-09-19</li>
<li>Automatic offers: </li>
<ul>
<li>rayane-djouah | Reviewer | 104113003</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @Christinadobrzyn</details>

**Comment by IuliiaHerets** [comment]
@Christinadobrzyn  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by ChavdaSachin** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Task - Long Task title is overlapped by task arrow

### What is the root cause of that problem?
- `ArrowRight` icon is not included in view container. 
https://github.com/Expensify/App/blob/4c424a2b6efec6aef7477adc4e22a1b7460947c5/src/components/ReportActionItem/TaskPreview.tsx#L125-L129

### What changes do you think we should make in order to solve the problem?
Move `icon` in view container.
```js
                <View style={[styles.flex1, styles.flexRow, styles.alignItemsStart, styles.mt1]}>
                    <View style={[styles.taskCheckboxWrapper, styles.alignSelfCenter]}>
                        <Checkbox
                            style={[styles.mr2]}
                            isChecked={isTaskCompleted}
                            disabled={!Task.canModifyTask(taskReport, currentUserPersonalDetails.accountID)}
                            onPress={Session.checkIfActionIsAllowed(() => {
                                if (isTaskCompleted) {
                                    Task.reopenTask(taskReport);
                                } else {
                                    Task.completeTask(taskReport);
                                }
                            })}
                            accessibilityLabel={translate('task.task')}
                        />
                    </View>
                    {taskAssigneeAccountID > 0 && (
                        <Avatar
                            containerStyles={[styles.mr2, styles.alignSelfCenter, isTaskCompleted ? styles.opacitySemiTransparent : undefined]}
                            source={avatar}
                            size={CONST.AVATAR_SIZE.SMALL}
                            avatarID={taskAssigneeAccountID}
                            type={CONST.ICON_TYPE_AVATAR}
                        />
                    )}
                    <View style={[styles.alignSelfCenter]}>
                        <RenderHTML html={isTaskCompleted ? `<completed-task>${htmlForTaskPreview}</completed-task>` : htmlForTaskPreview} />
                    </View>
                <Icon
                    src={Expensicons.ArrowRight}
                    fill={StyleUtils.getIconFillColor(getButtonState(isHovered))}
                />

                </View>
```
### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by Christinadobrzyn** [comment]
I can reproduce this. I think it can be external and I also think we need design to review this to see what would be best for the layout of the arrow.

**Comment by dannymcclain** [comment]
I'm pretty sure this is a regression. Not sure from where but this definitely used to not happen. We just need to wrap the text to multiple lines like we used to.

**Comment by Christinadobrzyn** [comment]
Ah thanks @dannymcclain - let me dig into any PRs that might relate...

**Comment by Christinadobrzyn** [comment]
Hey @thienlnam this is a regression that was found when executing this [GH](https://github.com/Expensify/Expensify/issues/294769). Do you happen to know which PR this might relate to?

**Comment by thienlnam** [comment]
I'm not sure, maybe this one? https://github.com/Expensify/App/pull/48552

**Comment by Christinadobrzyn** [comment]
@rayane-djouah would you be able to help us look into what this might be a regression of? Or, I think if we can't find the original PR we should consider just fixing this as a new job. @rayane-djouah let me know your thoughts

**Comment by bernhardoj** [comment]
Reverting https://github.com/Expensify/App/pull/48552 locally does fix the issue for me.

Inspecting the element on prod shows that the element structure matches the changes from the PR.
<img width="300" alt="image" src="https://github.com/user-attachments/assets/c81d056f-bf25-4ae1-9ff5-7f11c5294298">


A View (div) with an align self: center and a children of the RenderHTML.

**Comment by Gajendra-Gupta** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Task - Long Task title is overlapped by task arrow
### What is the root cause of that problem?
[App/src/components/ReportActionItem/TaskPreview.tsx](https://github.com/Expensify/App/blob/65a21fc7936b81e389be08c57d96b1089c8ff5c1/src/components/ReportActionItem/TaskPreview.tsx#L122-L124)

Lines 122 to 124 in [65a21fc](https://github.com/Expensify/App/commit/65a21fc7936b81e389be08c57d96b1089c8ff5c1)
```
<View style={[styles.alignSelfCenter]}> 
         <RenderHTML html={isTaskCompleted ? `<completed-task>${htmlForTaskPreview}</completed-task>` : htmlForTaskPreview} /> 
     </View>
```
### What changes do you think we should make in order to solve the problem?
We need to replace this code
```
<View style={[styles.alignSelfCenter]}> 
         <RenderHTML html={isTaskCompleted ? `<completed-task>${htmlForTaskPreview}</completed-task>` : htmlForTaskPreview} /> 
     </View>
```
by

```<RenderHTML html={isTaskCompleted ? `<completed-task>${htmlForTaskPreview}</completed-task>` : htmlForTaskPreview} />```

Also If you want arrow in center then need to add style `styles.alignItemsCenter` on this
https://github.com/Expensify/App/blob/65a21fc7936b81e389be08c57d96b1089c8ff5c1/src/components/ReportActionItem/TaskPreview.tsx#L87
so code will be 
````
<PressableWithoutFeedback
                onPress={() => Navigation.navigate(ROUTES.REPORT_WITH_ID.getRoute(taskReportID))}
                onPressIn={() => DeviceCapabilities.canUseTouchScreen() && ControlSelection.block()}
                onPressOut={() => ControlSelection.unblock()}
                onLongPress={(event) => showContextMenuForReport(event, contextMenuAnchor, chatReportID, action, checkIfContextMenuActive)}
                shouldUseHapticsOnLongPress
                style={[styles.flexRow, styles.justifyContentBetween, styles.alignItemsCenter]}
                role={CONST.ROLE.BUTTON}
                accessibilityLabel={translate('task.task')}
            >
````
### What alternative solutions did you explore? (Optional)

**Comment by Christinadobrzyn** [comment]
Thanks @rayane-djouah and @bernhardoj - if this is a regression of https://github.com/Expensify/App/pull/48552 should someone on the PR provide a proposal (I'm not sure)

**Comment by shawnborton** [comment]
I think I like the idea of the `styles.flex1` solution, but with the top alignment that we have from the task view page. Curious if the other designers agree.

**Comment by dannymcclain** [comment]
Can we see a side-by-side of what Shawn is suggesting and everything being centered? I'm not in love with the alignment here: 
![CleanShot 2024-09-18 at 08 09 51@2x](https://github.com/user-attachments/assets/c73be7b3-a78d-4baf-bd51-1ad642bcfd70)

and would be curious to see what it's like with everything centered (on the preview _only_)

**Comment by ChavdaSachin** [comment]
Here you go
|current | icon centered |
|--------|---------|
|<img width="389" alt="Screenshot 2024-09-18 at 6 51 28 PM" src="https://github.com/user-attachments/assets/f33d4906-ae35-40fe-86a3-1568b5e9e405">|<img width="390" alt="Screenshot 2024-09-18 at 6 51 00 PM" src="https://github.com/user-attachments/assets/bf82d280-2850-42d2-bf5a-827251a5c1f6">|

**Comment by dannymcclain** [comment]
Thank you!

Of those options, centered definitely looks best to my eye. Would love to hear @Expensify/design's thoughts/if they have any other alignment ideas I might not be thinking of.

**Comment by dannymcclain** [comment]
Ah gotcha - yeah I do think that's what Shawn was suggesting. Let's see what him and @dubielzyk-expensify think!

I think I'm down with center-aligned or your "Updated style 2" - with a _teeny tiny_ preference for centered.

**Comment by shawnborton** [comment]
Can we see the center aligned version where it has 3+ lines, and where there are multiple tasks in a row?

**Comment by dubielzyk-expensify** [comment]
I think I prefer when it's aligned to the top-aligned, but I'd still prefer the checkbox to be vertically aligned to the avatar when that's present, and the text when there's no avatar present:

![image](https://github.com/user-attachments/assets/bb95f0d6-b2a0-41ad-a35c-f284d323cc9c)

If both @dannymcclain and @shawnborton prefer the centered version, I'm happy to go for that though

**Comment by shawnborton** [comment]
I definitely prefer Jon's latest mockups after seeing the centered version with more than two lines... I think the centered version has the potential to look a little wild when the task title is long, not to mention it's not common to see bulleted list items (I know we're using checkboxes, but similar story here) where the bullet can float in the middle of the row. So I think I am team align to top, with those small adjustments to make sure the checkbox + avatar + arrow feel vertically aligned.

**Comment by dannymcclain** [comment]
Great point about bulleted lists Shawn, I am absolutely down to go with that version that you and Jon are into. Makes perfect sense to me!

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Task title is out of bounds when it's very long. We also want to center the checkbox to the avatar.

### What is the root cause of that problem?
This happen when we wrap the RenderHTML with a view.
https://github.com/Expensify/App/blob/040c8017219d4ca3650e41d41660c04d6b5a2d64/src/components/ReportActionItem/TaskPreview.tsx#L122-L124

If we look at the inspect element, the render HTML elements has the flex 1, but the new view wrapper doesn't have it, so the width is not restricted.

### What changes do you think we should make in order to solve the problem?

To fix the out of bound long text, we can add flex 1 to the view wrapper.
https://github.com/Expensify/App/blob/040c8017219d4ca3650e41d41660c04d6b5a2d64/src/components/ReportActionItem/TaskPreview.tsx#L122-L124

to center the checkbox to the avatar and the text, remove the align self center style from here
https://github.com/Expensify/App/blob/040c8017219d4ca3650e41d41660c04d6b5a2d64/src/components/ReportActionItem/TaskPreview.tsx#L114-L115
https://github.com/Expensify/App/blob/040c8017219d4ca3650e41d41660c04d6b5a2d64/src/components/ReportActionItem/TaskPreview.tsx#L98

It will make the checkbox centered to the text thanks to `taskCheckboxWrapper` style which has the same height as the text height and centered vertically. But for the avatar, since the size is bigger, we need a bigger height too. So, we need to update `taskCheckboxWrapper` to accept a height param (or move it to StyleUtils).

```
taskCheckboxWrapper: (height) => ({
    height,
    ...flex.justifyContentCenter,
}),
```

Then, pass the avatar height if there is an avatar.

```
const checkboxWrapperHeight = taskAssigneeAccountID > 0 ? StyleUtils.getAvatarSize(CONST.AVATAR_SIZE.SMALL) : variables.fontSizeNormalHeight;

<View style={[styles.taskCheckboxWrapper(checkboxWrapperHeight)]}>
    <Checkbox ...
```

If we want to center the arrow too, we can add the same style to the icon
https://github.com/Expensify/App/blob/040c8017219d4ca3650e41d41660c04d6b5a2d64/src/components/ReportActionItem/TaskPreview.tsx#L126-L129

```
additionalStyles={styles.taskCheckboxWrapper(checkboxWrapperHeight)}
```

<img width="425" alt="image" src="https://github.com/user-attachments/assets/7d055246-9736-4eae-9db4-ebd84385241e">

**Comment by ChavdaSachin** [comment]
## proposal 
[updated](https://github.com/Expensify/App/issues/49086#issuecomment-2346407869)

**Comment by bernhardoj** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/49086#issuecomment-2361077793)

**Comment by Christinadobrzyn** [comment]
Hi @rayane-djouah, when you have a moment can you check out the revised proposals? TY!

**Comment by Christinadobrzyn** [comment]
head up - I'm going to be ooo for the rest of the week. I don't think there's any reason to add another BZ teammate as we're reviewing proposals. 

@arosiclair feel free to review this proposal review when you have a moment! TY! https://github.com/Expensify/App/issues/49086#issuecomment-2368990363

**Comment by arosiclair** [comment]
> > We can modify the style of the html renderer, but instead of doing that, we can just replace the RenderHTML and its View wrapper with a simple Text component.
>
> Note: We introduced the use of RenderHTML in the task preview in [this commit](https://github.com/Expensify/App/commit/f4e6fb0c8a337f0c9b5600a1ad0f6b4836536f38) to display the user mention. However, since we are now displaying the user photo instead of mention, RenderHTML is no longer needed, we can use Text instead 👍 .

I had a concern about this as well. @rayane-djouah @bernhardoj can we be sure to test that scenario for regressions? Thanks.

**Comment by bernhardoj** [comment]
PR is ready

cc: @rayane-djouah

**Comment by bernhardoj** [comment]
The PR was reverted because of https://github.com/Expensify/App/issues/50209. In that issue, the task name is not updated dynamically.

The root cause is, the `react-compiler` automatically memoize the task title here. It will only be updated if `taskReportID` or `action?.childReportName` is updated.
https://github.com/Expensify/App/blob/99f280b0edae75ff59614a1c5e98c47b39800aa5/src/components/ReportActionItem/TaskPreview.tsx#L72

We don't touch the task title code, but why this happens after our PR?

Before the PR, `react-compiler` somehow wrongly optimized the component by causing the task title to be recomputed on every re-render. So, even if we write:
```
const taskTitle = useMemo(() => Str.htmlEncode(TaskUtils.getTaskTitle(taskReportID, action?.childReportName ?? '')), []); // empty deps
```

the memo will still run on every re-render. If we disable the `react-compiler` for that component by using "use no memo" directive, then the `useMemo` will work normally again.

After the PR, `react-compiler` optimization now works as expected.

The new PR revert the "revert" and add the fix for above issue by passing the `taskReport` instead of its ID.

cc: @rayane-djouah

**Comment by Christinadobrzyn** [comment]
A few more days until payment but creating a summary in preparation

Payouts due:

- [ ] Contributor: $250 @bernhardoj (NewDot - please submit a request for payment)
- [ ] Reviewer: $250 @rayane-djouah (Upwork offer - https://www.upwork.com/nx/wm/offer/104113003)

**Comment by bernhardoj** [comment]
Requested in ND.

**Comment by Christinadobrzyn** [comment]
Hum, @rayane-djouah @bernhardoj can you confirm if this is a regression? https://github.com/Expensify/App/issues/50209

Or what needs to be resolved before payment?

**Comment by bernhardoj** [comment]
I wouldn't call it a regression since the root cause of the issue exists before https://github.com/Expensify/App/issues/50209 which I explained [here](https://github.com/Expensify/App/issues/49086#issuecomment-2395057167).

cc: @arosiclair

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj (agree with the comment above)

**Comment by arosiclair** [comment]
Agree as well

**Comment by Christinadobrzyn** [comment]
okay thanks @bernhardoj and @arosiclair!

Paid out based on this payment summary - https://github.com/Expensify/App/issues/49086#issuecomment-2409673613
Regression test - https://github.com/Expensify/Expensify/issues/436505

Closing as complete!
